packages/typescript/action-kit/src/built-in-actions/download-app.ts

```diff
@@ -28,28 +28,33 @@ export async function downloadApp(
 
   const extension = getExtension(DOGU_DEVICE_PLATFORM);
   printable.info('Get application list for', {
-    version: appVersion,
+    version: by.appVersion,
+    packageName: by.appPackageName,
     extension,
   });
 
-  let applications: Application[];
-  if (appVersion === 'latest') {
-    const { applications: apps } = await consoleActionClient.getApplicationList({
-      latestOnly: true,
+  let applications: Application[] = [];
+  if (by.appPackageName) {
+    const { applications: apps } = await consoleActionClient.getApplicationsWithUniquePackage({
       extension,
     });
     applications = apps;
-  } else {
+  } else if (by.appVersion) {
     const { applications: apps } = await consoleActionClient.getApplicationList({
-      version: appVersion,
+      version: by.appVersion,
       extension,
     });
     applications = apps;
   }
 
   if (applications.length === 0) {
-    throw new Error(`No application found for version ${appVersion} and extension ${extension}`);
+    throw new Error(
+      `No application found for ${
+        by.appPackageName ? `appPackageName ${by.appPackageName}` : by.appVersion ? `appVersion ${by.appVersion}` : 'empty condition'
+      } and extension ${extension}`,
+    );
   }
+
   const application = applications[0];
```

Comment:
Shouldn't we filter by.appPackageName among the app list obtained from consoleActionClient.getApplicationsWithUniquePackage?

Reply:
👍